receiver/prometheusreceiver/internal/metricfamily.go

```diff
@@ -359,3 +363,27 @@ func populateLabelValues(orderedKeys []string, ls labels.Labels) []*metricspb.La
 	}
 	return lvs
 }
+
+// TODO: add support for other resource types
+func createNodeResource(resourceLabels map[string]string) *resourcepb.Resource {
+	location, ok := resourceLabels[resourcekeys.CloudKeyZone]
```

Comment:
It is not clear why this resource labels get a special treatment in a generic Prometheus receiver. Please add a PR description to help understand the intent and the purpose of this change.

If this PR solves a particular problem please describe the problem in the PR or as a separate Issue and link to it from the PR.

Reply:
I think we should not use this specific NodeResource, but get the type as well from the config:

```
    resource:
      type: "host"
      labels: {
      "cloud.zone": "us-central-1a",
      "k8s.cluster.name": "prometheus-cluster",
      "host.name": "prometheus-node",
      }
```

Also we don't need to check for the label keys here. We just put what we get from the config.

Reply:
+1 to this. Stackdriver exporter knows how to map the generic node/host resources to the ones that Stackdriver supports.

Reply:
sgtm - done